contracts/stake/src/tests/bond.rs

```diff
@@ -226,3 +226,57 @@ fn unbond_wrong_user_stake_not_found() {
 
     staking.unbond(&user2, &10_000, &2_000);
 }
+
+#[test]
+fn pay_rewards_during_unbond() {
+    const AMOUNT: i128 = 100_000;
+    const WITHDRAW_TIMESTAMP: u64 = 2_000;
+
+    let env = Env::default();
+    env.mock_all_auths();
+
+    let admin = Address::generate(&env);
+    let user = Address::generate(&env);
+    let manager = Address::generate(&env);
+    let owner = Address::generate(&env);
+
+    let lp_token = deploy_token_contract(&env, &admin);
+    let staking = deploy_staking_contract(&env, admin.clone(), &lp_token.address, &manager, &owner);
+
+    lp_token.mint(&user, &AMOUNT);
+    lp_token.mint(&owner, &AMOUNT);
+
+    staking.create_distribution_flow(&owner, &lp_token.address);
+    staking.fund_distribution(&owner, &0u64, &10_000u64, &lp_token.address, &AMOUNT);
+
+    env.ledger().with_mut(|li| {
+        li.timestamp = WITHDRAW_TIMESTAMP;
+    });
```

Comment:
You created a distribution ranging from timestamp 0 to timestamp 10_000, and you immediately jump 2000 seconds ahead. Why?

Reply:
I chose numbers based on other places in our code where we change the ledger().timestamp. Do the numbers we use matter?